File: template-rest-assured-java/templates/src/test/java/{{package_name}}/tests/CreateUserTests.java
package {{packame_name}}.tests;

import com.github.javafaker.Faker;
import io.restassured.response.ValidatableResponse;
import org.junit.Test;

import static {{packame_name}}.service.UserService.*;
import static {{packame_name}}.support.utils.BaseRest.*;
import static {{packame_name}}.support.utils.GetVariable.getProp;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class CreateUserTests {
    ValidatableResponse response;

    static Faker faker = new Faker();
    static String name = faker.name().firstName();
    static String email = faker.internet().emailAddress();

    @Test
    public void createUserWithSuccess(){
        createUser(name, email, "123456","false");

        response = getResponse().then().log().all();

        assertEquals(201,response.extract().statusCode());
        assertEquals(getProp("message"), response.extract().path("message"));
        assertNotNull(response.extract().path("_id"));

    }
}

File: template-rest-assured-kotlin/templates/src/test/kotlin/{{packame_name}}/tests/CreateUserTests.kt
package {{packame_name}}.tests

import {{packame_name}}.service.UserService.createUser
import io.restassured.response.ValidatableResponse
import com.github.javafaker.Faker
import org.junit.Assert
import org.junit.Test
import {{packame_name}}.support.utils.BaseRest.response
import {{packame_name}}.support.utils.GetVariable.getProp

class CreateUserTests {
    var responseRequest: ValidatableResponse? = null
    @Test
    fun createUserWithSuccess() {
        createUser(name, email, "123456", "false")
        responseRequest = response!!.then().log().all()
        Assert.assertEquals(201, responseRequest!!.extract().statusCode().toLong())
        Assert.assertEquals(getProp("message"), responseRequest!!.extract().path("message"))
        Assert.assertNotNull(responseRequest!!.extract().path("_id"))
    }

    companion object {
        var faker = Faker()
        var name = faker.name().firstName()
        var email = faker.internet().emailAddress()
    }
}